src/modules/launcher/PowerLauncher/Helper/EnvironmentHelper.cs

```diff
@@ -5,88 +5,215 @@
+                    // Initialize variables for length of environment variable name and value. Using this variables prevent us from null value exceptions.
+                    // => We added this because of the issue #13172 where a user reported System.ArgumentNullException from "Environment.SetEnvironmentVariable()".
+                    int varNameLength = kv.Key == null ? 0 : kv.Key.Length;
+                    int varValueLength = kv.Value == null ? 0 : kv.Value.Length;
+
+                    // The name of environment variables must not be null, empty or have a length of zero.
+                    // But if the value of the environment variable is null or an empty string then the variable is explicit defined for deletion. => Here we don't need to check anything.
+                    // => We added the if statement (next line) because of the issue #13172 where a user reported System.ArgumentNullException from "Environment.SetEnvironmentVariable()".
+                    if (!string.IsNullOrEmpty(kv.Key) & varNameLength > 0)
+                    {
+                        try
+                        {
+                            // If the variable is not listed as protected/don't override on process level, then update it. (See method "GetProtectedEnvironmentVariables" of this class.)
+                            if (!_protectedProcessVariables.Contains(kv.Key))
+                            {
+                                // We have to delete the variables first that we can update their name if changed by the user. (Example: "path" => "Path")
+                                // The machine and user variables that have been deleted by the user having an empty string as variable value. Because of this we check the values of the variables in our dictionary against "null" and "string.Empty". This check prevents us from invoking a (second) delete command.
+                                // The dotnet method doesn't throw an exception if the variable which should be deleted doesn't exist.
+                                Environment.SetEnvironmentVariable(kv.Key, null, EnvironmentVariableTarget.Process);
+                                if (!string.IsNullOrEmpty(kv.Value))
+                                {
+                                    Environment.SetEnvironmentVariable(kv.Key, kv.Value, EnvironmentVariableTarget.Process);
+                                }
+                            }
+                            else
+                            {
+                                Log.Warn($"Skipping update of the environment variable [{kv.Key}] for the PT Run process. This variable is listed as protected process variable. (The variable value has a length of [{varValueLength}].)", typeof(PowerLauncher.Helper.EnvironmentHelper));
```

Comment:
@jaimecbernardo 
Should we use warning level or is info level the better choice here?

Reply:
If I understood correctly, this is when something tries to change an environment variable that we got from the process level, not the machine or user level. I think the change of such a environment variable warrants a warning level, as it seems like it'd be weird behavior.

Reply:
I will ad an note to the error message: "Changing such a variable can cause unexpected behavior."